### PR TITLE
Step 68 simplex

### PR DESCRIPTION
**Pull request by blaisb**
First try at a Simplex version of step-68 on a fully distributed triangulation. After some bug fixing, I still get an error when the mesh is very fine, but is now really deep down inside the library. I am still trying to find a solution. The assertion thrown I get is:

```
An error occurred in line <2808> of file </home/blaisb/work/dealii_main/dealii/source/grid/tria_accessor.cc> in function
    bool dealii::CellAccessor<dim, spacedim>::at_boundary(unsigned int) const [with int dim = 2; int spacedim = 2]
The violated condition was: 
    static_cast<typename ::dealii::internal::argument_type<void( typename std::common_type<decltype(i), decltype(this->n_faces())>::type)>:: type>(i) < static_cast<typename ::dealii::internal::argument_type<void( typename std::common_type<decltype(i), decltype(this->n_faces())>::type)>:: type>(this->n_faces())
Additional information: 
    Index 3 is not in the half-open range [0,3).

Stacktrace:
#0  /home/blaisb/work/dealii_main/inst/lib/libdeal_II.g.so.9.3.0-pre: dealii::CellAccessor<2, 2>::at_boundary(unsigned int) const
#1  /home/blaisb/work/dealii_main/inst/lib/libdeal_II.g.so.9.3.0-pre: std::vector<dealii::Triangulation<2, 2>::active_cell_iterator, std::allocator<dealii::Triangulation<2, 2>::active_cell_iterator> > dealii::GridTools::find_cells_adjacent_to_vertex<2, dealii::Triangulation, 2>(dealii::Triangulation<2, 2> const&, unsigned int)
#2  /home/blaisb/work/dealii_main/inst/lib/libdeal_II.g.so.9.3.0-pre: std::pair<dealii::Triangulation<2, 2>::active_cell_iterator, dealii::Point<2, double> > dealii::GridTools::find_active_cell_around_point<2, dealii::Triangulation, 2>(dealii::Mapping<2, 2> const&, dealii::Triangulation<2, 2> const&, dealii::Point<2, double> const&, std::vector<bool, std::allocator<bool> > const&, double)
#3  /home/blaisb/work/dealii_main/inst/lib/libdeal_II.g.so.9.3.0-pre: dealii::Particles::ParticleHandler<2, 2>::sort_particles_into_subdomains_and_cells()
#4  ./step-68: Step68::ParticleTracking<2>::run()
#5  ./step-68: main


```
The step-68 simplex is currently set-up with parameters that will break it. The finer the background mesh is, the easier it is to break. I will continue debugging that, but if @gassmoeller , @luca-heltai, @bangerth or @peterrum  you have ideas, please tell me :).

### CODE REVIEW

**Comment by peterrum** [comment]
I think this lines causes the trouble:

https://github.com/dealii/dealii/blob/30eb1dfdeac12284f9899478aa5f457edbf95c56/source/grid/grid_tools.cc#L1550-L1551

**Comment by blaisb** [comment]
> I think this lines causes the trouble:
> 
> https://github.com/dealii/dealii/blob/30eb1dfdeac12284f9899478aa5f457edbf95c56/source/grid/grid_tools.cc#L1548-L1551
> 
> I can write a new utility function with a similar functionality tomorrow.

Take you time. I just felt that this was getting too deep outside of particles, hence me opening the PR as a WIP.

**Comment by peterrum** [comment]
Should work now. I will create a separate PR for the new function tomorrow.

**Comment by peterrum** [comment]
@blaisb You can now rebase and throw away my last two commits. Thx!

**Comment by blaisb** [comment]
@peterrum  I finalised the step-68 simplex test. It is slightly different than the real step-68 in order to make it run faster, produce less files and be reproducible with more reliable output. I have kept some of the comments but I can remove them if needed :)

File: tests/simplex/step-68.cc [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE.md at
+ * the top level directory of deal.II.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Authors: Bruno Blais, Toni El Geitani Nehme, Rene Gassmoeller, Peter Munch
```
Comment (peterrum) [inline tests/simplex/step-68.cc:17]: Could you shortly describe the made chantes?

**Review by peterrum — APPROVED** [review]
Looks good! Thanks. May I ask you document the made changes and to squash the commits.

**Comment by blaisb** [comment]
@peterrum  I rebased and I cleaned the comments. I added a small list that describes the core differences between this test and the step68 :).

**Pull request merged**